test_expect_failure/dep_analyzer_modes/BUILD

```diff
@@ -0,0 +1,56 @@
+package(default_visibility = ["//visibility:public"])
+load("//scala:scala.bzl", "scala_library", "scala_binary", "scala_test")
+
+scala_library(
+    name="error_mode",
+    dependency_analyzer_mode="error",
+    srcs=[
```

Comment:
Rather than error I think, "strict" or something reads better. mode = error seems weird to me. Maybe on_missing_dependency = "error" | "warn" | "default_failure" is bit more ergonomic. Default failure means what you have "off" meaning.

To users, we have strict dependencies in the error or off mode, which seems a bit weird to me. Calling this analyzer mode is a bit too much about the implementation and not what users care about in my view.